src/mscorlib/shared/System/IO/Path.Unix.cs

```diff
@@ -191,12 +191,7 @@ public static string GetPathRoot(string path)
 
         public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
         {
-            ReadOnlySpan<char> result = new ReadOnlySpan<char>();
-
-            if (path == null)
-                return result;
-
-            if (PathInternal.IsEffectivelyEmpty(path))
+            if (PathInternal.IsEffectivelyEmpty(new string(path.ToArray())))
```

Comment:
You should just check `path.Length == 0` instead of allocating a new string just for the check.